staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -123,6 +123,18 @@ const (
 	// ErrorTypeInternal is used to report other errors that are not related
 	// to user input.  See InternalError().
 	ErrorTypeInternal ErrorType = "InternalError"
+	// TooLongFailCode is for MaxLengthExceed error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	TooLongFailCode ErrorType = "MaxLengthExceed"
+	// MaxItemsFailCode is for MaxItemsExceed error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	MaxItemsFailCode ErrorType = "MaxItemsExceed"
+	// TooManyPropertiesCode is for MaxPropertiesExceed error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	TooManyPropertiesCode ErrorType = "MaxPropertiesExceed"
+	// InvalidTypeCode is for type invalid error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	InvalidTypeCode ErrorType = "TypeInvalid"
```

Comment:
Suggest following the existing format of `FieldValue<X>`, maybe `FieldValueTypeInvalid`, and update the docs to indicate it means the value did not match the schema type for that field. Omit CEL-specific docs from this package

There's also a comment at the top of this block that `These values are duplicated in api/types.go` ... do we need to duplicate this new value there?

Reply:
Only first few are duplicated in [api/types](https://github.com/kubernetes/kubernetes/blob/221b60fb82fc0a6de3034e342de99223a04585bb/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L972) plus we only use the added err for cr validation to omit cel rules from running. I guess it should be fine to not update in api/types